modules/dnn/src/webgpu/src/tensor.cpp

```diff
@@ -82,36 +83,25 @@ Tensor Tensor::reshape(const void* data, const std::vector<int>& shape,
     if (alloc || new_size > size_in_byte_)
         alloc = true;
     size_in_byte_ = new_size;
-    if (alloc || !buffer_)
+    if(alloc)
     {
-        // TODO: specific data type 
         buffer_.reset(new Buffer(device_, data, size_in_byte_, usage_));
+        return * this;
     }
-    else if (data)
-    {
-        buffer_->setBufferData(data, size_in_byte_);
-    }
-    return *this;
+    fillData(data);
 }
 
-Tensor Tensor::setUniform(const void * data, Format fmt) 
+Tensor Tensor::fillData(const void * data) 
 {
-    if (device_ == nullptr)
-    {
-        CV_Error(Error::StsError, "device is NULL");
-        return *this;
-    }
-    if (checkFormat(fmt) && fmt != format_) format_ = fmt;
     if (!buffer_)
     {
-        // TODO: specific data type 
         buffer_.reset(new Buffer(device_, data, size_in_byte_, usage_));
     }
```

Comment:
if ( size > size_in_byte_ ) , don't fill, but warning and return

Reply:
We just set size_in_byte_ (using the alloc flag to reset buffer) before using fillData, do we need to check again？

Reply:
ok, ignore this comment.